stdlib/list.gr

```diff
@@ -329,3 +329,54 @@ let join = (separator: String, items: List<String>) => {
     Some(s) => s,
   }
 }
+
+/**
+ * Reverses the list and appends items to the end
+ * 
+ * @param items: Items to be appended to the reversed list
+ * @param list: The list that is being reversed and appended to
+ * @since v0.4.5
+ */
+let revAppend = (items, list) => { 
+  append(reverse(list), items)
+}
```

Comment:
Sorry if I was a bit unclear. I wanted this so we could have an efficient implementation. Usually the main list goes last, but for `append`-like functions it goes in the order they get appended:

```suggestion
 * @param list1: The list to reverse
 * @param list2: The list to append
 * @since v0.4.5
 */
let rec revAppend = (list1, list2) => { 
  match (list1) {
    [hd, ...tl] => revAppend(tl, [hd, ...list2]),
    [] => list2
  }
}
```

We'll have to update the call sites to reverse the arguments as well.